compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/SemanticModel.java

```diff
@@ -57,6 +58,15 @@
      */
     List<Symbol> moduleLevelSymbols();
 
+    /**
+     * Retrieves the type of the expression in the specified text range. If it's not a valid expression, returns null.
```

Comment:
Shall we mention if the range is not valid, we return an empty optional? Instead of returns null